### PR TITLE
ISSUE-7 - Implement a PHP Framework for Sitepod

### PR DESCRIPTION
**Pull request by sigee**
#7

### CODE REVIEW

**Review by lewismc — COMMENTED** [review]
WOW... I would really like to have a look at this. If thats OK @sigee

File: src/Sitepod/Controller/Home.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        if (is_null($SETTINGS[PSNG_LASTMOD])) {
+            $SETTINGS[PSNG_LASTMOD] = PSNG_LASTMOD_FILEDATE;
+        }
+        if (is_null($SETTINGS[PSNG_LASTMOD_TIMEFORMAT])) {
+            $SETTINGS[PSNG_LASTMOD_TIMEFORMAT] = PSNG_TIMEFORMAT_LONG;
+        }
+
+        $layout = '';
+        /* It works because in startup.php you added it to the include_path */
+        /** @TODO: Change it to a template system Eg.: Twig (https://twig.sensiolabs.org/) */
```
Comment (sigee) [inline src/Sitepod/Controller/Home.php:152]: Fatfree also have templating, but poorly documented.

**Comment by lewismc** [comment]
Hi @sigee I just fetched this pull request and upon copying it into my Webserv I get the following

screen shot 2017-06-19 at 1 34 58 pm

Any ideas what is wrong?

**Comment by sigee** [comment]
Hi @lewismc, It is because I added a new dependency at the composer.json. You need to run 
```bash
 composer install
```
to download the Fatfree framework that is the new dependency.

**Comment by lewismc** [comment]
Thank you @sigee I though this. OK, so I've tested this out and it looks good. Can you explain a bit about how the new framework changes the design? I have never used FatFree before so I am keen to learn. Thanks for the work again...

**Comment by sigee** [comment]
@lewismc,
This was the first time when I used fatfree, so I'm also a newbie with it.
The biggest change is that, we use the fatfree router to call the correct controller / action instead of having a switch. The router parses the url (including method, that could be HEAD, GET, POST, PUT, DELETE, etc.) to find the correct route.
The switch cases were moved to controller classes into actions.
I'd create an [MVC structure](https://daveh.io/user/pages/01.blog/the-model-view-controller-pattern/the-mvc-pattern.png). At the moment the controllers are created, but there is no models and no view. View is everywhere. There are a lot of print in the code base. We should reorganise them.
It would be nice to use a templateing system to generate view. Eg.: [Fatfree template](https://fatfreeframework.com/3.6/views-and-templates)
If you have any question, don't hesitate to ask me.

**Comment by lewismc** [comment]
I am +1 for merging this. If you are able to work on the templating, then please progress with that on a separate pull request.
It would also be nice to have a command line interface or REST API to call Sitepod... what are your thoughts on that?

**Pull request merged**